mathics/core/element.py

```diff
@@ -444,6 +444,18 @@ def get_sequence(self):
     def get_string_value(self):
         return None
 
+    @property
+    def is_literal(self) -> bool:
+        """
+        True if the value can't change, i.e. a value is set and it does not
+        depend on definition bindings. That is why, in contrast to
+        `is_uncertain_final_definitions()`, we don't need a `definitions`
+        parameter.
+
+        Each subclass should decide what is right here.
+        """
+        raise NotImplementedError
```

Comment:
OK, but at least ``Number``, ``String`` and  ``ByteArrayAtom`` should implement it and return ``True``.